java/ql/test/library-tests/object-tostring-flow-binding-patterns/Test.java

```diff
@@ -0,0 +1,29 @@
+public class Test {
+
+  interface Intf { }
+  static class Specific implements Intf { public String toString() { return "Specific"; } }
+  static class Alternative implements Intf { public String toString() { return "Alternative"; } }
+
+  public static String caller() {
+
+    Alternative a = new Alternative(); // Instantiate this somewhere so there are at least two candidate types in general
+    return test(new Specific());
+
+  }
+
+  public static String test(Object o) {
+
+    if (o instanceof Object o2) {
```

Comment:
Should this have been s/`Object`/`Intf`/?

Reply:
Ah wait, I guess not, as that would probably count as being a too concrete type and hence not make the ObjectFlow lib kick in.